### PR TITLE
Use clap for argument parsing.

### PR DESCRIPTION
**Pull request by plietar**
The various binaries are consolidated into a single `outpack` command, and their command line arguments are unified through the use of the clap crate.

Some of the error types around the crate are cleaned up a bit to allow this consolidation. In particular the api and init modules now use an `anyhow` error, and the query module uses a `thiserror` to keep a bit more precision. There are still a lot of (ab)uses of `io::Error` around, but these can be updated later.

### CODE REVIEW

File: src/bin/outpack/args.rs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    /// Search for a packet in a repository
+    Search {
+        #[arg(short, long)]
+        root: String,
+        query: String,
+    },
+    /// Parse an outpack query, without evaluating it
+    Parse { query: String },
+    /// Start the outpack API server
+    ApiServer {
```
Comment (r-ash) [inline src/bin/outpack/args.rs:31]: How about just `Start`?

File: src/bin/outpack/args.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        path: String,
+
+        #[arg(long)]
+        path_archive: Option<String>,
+
+        #[arg(long)]
+        use_file_store: bool,
+
+        #[arg(long)]
+        require_complete_tree: bool,
```
Comment (r-ash) [inline src/bin/outpack/args.rs:20]: Can these have docs on them too?

**Review by r-ash — CHANGES_REQUESTED** [review]
This looks so nice, really tidy. clap looks really simple and error tidyup looks great.

Can you update the README? Has some usage docs in it with the current non-clap CLI.

File: src/bin/outpack/args.rs (hunk 1 above)
Comment (plietar) [reply src/bin/outpack/args.rs:31]: The command is called `outpack`, so `outpack start` seems pretty ambiguous IMO.
Alternatively we could keep the old `outpack_server` command name, but then it makes the other commands a bit odd.

Comment (plietar) [reply src/bin/outpack/args.rs]: Gone with `StartServer`.

File: Cargo.toml [hunk 3]
```diff
@@ -23,6 +23,9 @@ pest = "2.7"
 pest_derive = "2.7"
 itertools = "0.10.5"
 tempfile = "3.6.0"
+clap = { version = "4.4.8", features = ["derive"] }
```
[the remark this answers is not in the document]
Comment (plietar) [reply Cargo.toml:25]: Good catch. Thanks.

**Pull request merged**